### PR TITLE
add write constraint range check

### LINKED ISSUE #232: Check writeConstraint range
The range of values for a enumeration is checked against a field size. But there is no such check for a range. It is possible to assign a range that out of bounds field.

### PR DESCRIPTION
**Pull request by kossnikita**
fixes #232

### CODE REVIEW

**Review by burrbull — APPROVED** [review]

**Comment by burrbull** [comment]
rustfmt, please

File: svd-rs/src/writeconstraint.rs [hunk 1]
```diff
@@ -25,3 +27,22 @@ pub struct WriteConstraintRange {
...
+    OutOfRange(u64, core::ops::Range<u64>),
+}
+
+impl WriteConstraintRange {
+    pub(crate) fn check_range(&self, range: core::ops::Range<u64>) -> Result<(), SvdError> {
+        for v in [&self.min, &self.max] {
+            if !range.contains(v) {
+                return Err(Error::OutOfRange(*v, range.clone()).into());
+            }
+        }
```
Comment (burrbull) [inline svd-rs/src/writeconstraint.rs:45]: What sense of loop here? It can be done much faster with just `if`.

**Review by burrbull — APPROVED** [review]

**Comment by burrbull** [comment]
Possibly check also `self.min<=self.max`?

**Comment by kossnikita** [comment]
> `self.min<=self.max`?

I'm not sure they can't be equal.

File: svd-rs/src/writeconstraint.rs [hunk 2]
```diff
@@ -25,3 +27,28 @@ pub struct WriteConstraintRange {
...
+    #[error("Value {0} out of range {1:?}")]
+    OutOfRange(u64, core::ops::Range<u64>),
+    /// Minimum is greater than maximum.
+    #[error("Range minimum {0} is greater than maximum {1}")]
+    ReversedRange(u64, u64),
+}
+
+impl WriteConstraintRange {
+    pub(crate) fn check_range(&self, range: core::ops::Range<u64>) -> Result<(), SvdError> {
+        if &self.min > &self.max {
```
Comment (burrbull) [inline svd-rs/src/writeconstraint.rs:44]: I don't see sense in references here. Could you run `cargo clippy`, please.
Looks like I missed it in CI.

Comment (kossnikita) [reply svd-rs/src/writeconstraint.rs:44]: Yeah, you right

**Pull request merged**